numeric.h

```diff
@@ -0,0 +1,34 @@
+#ifndef NUMBER_THEORY_NUMERIC_H
+#define NUMBER_THEORY_NUMERIC_H
+
+#include <limits>
+#include <numeric>
+#include <utility>
+
+/**
+ * Common numeric functions.
+ */
+
+namespace number_theory {
```

Comment:
I wonder if it would be better to gather everything under namespace `tql` instead.

Reply:
Because tqlib may have many different categories of algorithms, it is better to isolate them with namespaces. "number_theory" is long as a name, but we can't find a better name yet.